core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedDocServicePlugin.java

```diff
@@ -156,8 +157,10 @@ private static void addMethodInfo(Map<Class<?>, Set<MethodInfo>> methodInfos,
         final Method method = service.method();
         final String name = method.getName();
         final TypeSignature returnTypeSignature =
-                isSuspendingFunction(method) ? toTypeSignature(kFunctionGenericReturnType(method))
-                                             : toTypeSignature(method.getGenericReturnType());
+                isKFunction(method) && !isReturnTypeNothing(method) ? toTypeSignature(
+                        kFunctionGenericReturnType(method))
+                                                                    : toTypeSignature(
+                        method.getGenericReturnType());
```

Comment:
Could you introduce a private method for readability?

Reply:
Updated